placeSpecificContent/theSpot/moneyTree/ACScripts/ACmoneyTree.js

```diff
@@ -0,0 +1,806 @@
+                }
+            }
+            if (targetGiver.username !== userList[userListGiverIndex].username){// Try not to pick the same giver twice in a row
+                targetGiver = userList[userListGiverIndex];
+            } else {
+                userListGiverIndex = randInt(0, userList.length-1);
+                targetGiver = userList[userListGiverIndex];
+            }
+        } else { // If there was no previous giver, then this is the first one.
+            targetGiver = userList[userListGiverIndex];
+        }
+    }
+}
+
+
+// This function picks the potential recipients of the present and qualified users.
+function pickRecipients(){
+    targetRecipients = [];
+    tempList = [];
+    tempList = userList.slice();// Make a copy of the user list to whittle down.
+    tempList.splice(userListGiverIndex, 1)[0]; // Do not include the giver on this list.
+    for (var i = 0; i < tempList.length; i++) {
+        if (tempList[i].staff === true) { // Remove all staff as they are not qualified to receive.
+            tempList.splice(i,1)[0];
+            i--;
+        }
+    }
+    if (tempList.length < RECIPIENT_MAX-1) { // If the list is less than 2 qualified receivers, then you don't have enough!
+        Entities.editEntity(marquisID, {text: "MOAR PPL PLZ"});
+        return;
+    }
+    var recipientCount = (tempList.length > RECIPIENT_MAX) ? RECIPIENT_MAX : tempList.length;
+    for (var j = 0; j < recipientCount; j++){ 
+        var index = randInt(0, tempList.length-1);
+        targetRecipients.push(tempList.splice(index-j, 1)[0]);// Add two or three recipients to the final list.     
+    }
+}
+
+
+// This function returns a random float between high and low limits.
+function randFloat(low, high) {
+    return low + Math.random() * (high - low);
+}
+
+
+// This function returns a random integer between high and low limits.
+function randInt(low, high) {
+    return Math.floor(randFloat(low, high));
+}
+
+
+// This function returns a gaussian random variate with the given mean and stdev.
+function gaussian(mean, stdev) {
+    var y2;
+    var useLast = false;
+    var y1;
+    if (useLast) {
+        y1 = y2;
+        useLast = false;
+    } else {
+        var x1, x2, w;
+        do {
+            x1 = 2.0 * Math.random() - 1.0;
+            x2 = 2.0 * Math.random() - 1.0;
+            w = x1 * x1 + x2 * x2;               
+        } while ( w >= 1.0);
+        w = Math.sqrt((-2.0 * Math.log(w))/w);
+        y1 = x1 * w;
+        y2 = x2 * w;
+        useLast = true;
+    }
+
+    var retval = mean + stdev * y1;
+    if (retval > 0) {
+        return retval;
+    }
+    return -retval;
+}
+
+
+// Send info to google sheets.
+function sendInput(recipientUsername, amount) {      
+    var paramString = encodeURLParams({
+        date: new Date().toLocaleString(),
+        recipientUsername: recipientUsername,
+        amount: amount
+    });
+
+    var request = new XMLHttpRequest();
```

Comment:
i'd normally say "switch this over to using the request module", but let's not bother this time.